packages/mobile/src/harmony-native/components/Comments/Timestamp/Timestamp.tsx

```diff
@@ -0,0 +1,44 @@
+import { Text } from '../..'
+
+import type { TimeUnit, TimestampProps } from './types'
+import {
+  DAY_IN_MONTH,
+  HR_IN_DAY,
+  MIN_IN_HR,
+  MONTH_IN_YEAR,
+  MS_IN_S,
+  S_IN_MIN
+} from './types'
+
+const timeUnitMsMap: Record<TimeUnit, number> = {
+  m: MS_IN_S * S_IN_MIN,
+  h: MS_IN_S * S_IN_MIN * MIN_IN_HR,
+  d: MS_IN_S * S_IN_MIN * MIN_IN_HR * HR_IN_DAY,
+  mo: MS_IN_S * S_IN_MIN * MIN_IN_HR * HR_IN_DAY * DAY_IN_MONTH,
+  y: MS_IN_S * S_IN_MIN * MIN_IN_HR * HR_IN_DAY * DAY_IN_MONTH * MONTH_IN_YEAR
+} as const
+
+const getTextFromTime = (time: Date) => {
+  const then = new Date(time).getTime()
+  const now = Date.now()
+  const diff = now - then
+  let unit: TimeUnit | null = null
+
+  Object.entries(timeUnitMsMap).forEach(([u, ms]) => {
+    if (diff >= ms) {
+      unit = u as TimeUnit
+    }
+  })
```

Comment:
nit: gains are minimal but it would make more sense to loop the reverse of timestamps and return as soon as a timestamp is found. 
Using your implementation above I think it would be like:

```suggestion
const timeUnitMsMap: Record<TimeUnit, number> = {
  y: MS_IN_S * S_IN_MIN * MIN_IN_HR * HR_IN_DAY * DAY_IN_MONTH * MONTH_IN_YEAR,
  mo: MS_IN_S * S_IN_MIN * MIN_IN_HR * HR_IN_DAY * DAY_IN_MONTH,
  d: MS_IN_S * S_IN_MIN * MIN_IN_HR * HR_IN_DAY,
  h: MS_IN_S * S_IN_MIN * MIN_IN_HR,
  m: MS_IN_S * S_IN_MIN
} as const

  Object.entries(timeUnitMsMap).some(([u, ms]) => {
    if (diff >= ms) {
      unit = u as TimeUnit
      return true // break loop early
    }
    return false // keep looping
  })
```

(technically I guess JS doesn't guarantee object.entries order but in practice I think it's always preserved, idk alternatively could use `Map()`)

Reply:
Also, this probably belongs in `common/utils`? iirc we're able to pull common into harmony?

Reply:
At minimum we should put it only in harmony proper and pull it into native from there instead of copy-paste